framework/test/unit/modules/dpos_v2/commands/pom.spec.ts

```diff
@@ -0,0 +1,784 @@
+				})
+				.createCommandExecuteContext<PomTransactionParams>(pomCommand.schema);
+
+			await expect(pomCommand.execute(context)).rejects.toThrow(
+				'Difference between header2.height and current height must be less than 260000.',
+			);
+		});
+
+		it('should throw error when header1 is not properly signed', async () => {
+			transactionParamsDecoded = {
+				header1: codec.encode(blockHeaderSchema, transactionParamsPreDecoded.header1),
+				header2: codec.encode(blockHeaderSchema, transactionParamsPreDecoded.header2),
+			};
+			transactionParams = codec.encode(pomCommand.schema, transactionParamsDecoded);
+			transaction.params = transactionParams;
+			context = testing
+				.createTransactionContext({
+					stateStore,
+					transaction,
+					header,
+				})
+				.createCommandExecuteContext<PomTransactionParams>(pomCommand.schema);
+
+			when(dposUtils.validateSignature as any)
+				.calledWith(
+					'LSK_BH_',
+					expect.any(Buffer),
+					expect.any(Buffer),
+					transactionParamsPreDecoded.header1.signature,
+					expect.any(Buffer),
+				)
+				.mockReturnValue(false);
+			when(dposUtils.validateSignature as any)
+				.calledWith(
+					'LSK_BH_',
+					expect.any(Buffer),
+					expect.any(Buffer),
+					transactionParamsPreDecoded.header2.signature,
+					expect.any(Buffer),
+				)
+				.mockReturnValue(true);
+
+			await expect(pomCommand.execute(context)).rejects.toThrow(
+				'Invalid block signature for header 1',
+			);
+		});
+
+		it('should throw error when header2 is not properly signed', async () => {
+			transactionParamsDecoded = {
+				header2: codec.encode(blockHeaderSchema, {
+					...transactionParamsPreDecoded.header2,
+					generatorAddress: transactionParamsPreDecoded.header2.generatorAddress,
+				}),
+				header1: codec.encode(blockHeaderSchema, transactionParamsPreDecoded.header1),
+			};
+			transactionParams = codec.encode(pomCommand.schema, transactionParamsDecoded);
+			transaction.params = transactionParams;
+			context = testing
+				.createTransactionContext({
+					stateStore,
+					transaction,
+					header,
+				})
+				.createCommandExecuteContext<PomTransactionParams>(pomCommand.schema);
+
+			when(dposUtils.validateSignature as any)
+				.calledWith(
+					'LSK_BH_',
+					expect.any(Buffer),
+					expect.any(Buffer),
+					transactionParamsPreDecoded.header1.signature,
+					expect.any(Buffer),
+				)
+				.mockReturnValue(true);
+			when(dposUtils.validateSignature as any)
+				.calledWith(
+					'LSK_BH_',
+					expect.any(Buffer),
+					expect.any(Buffer),
+					transactionParamsPreDecoded.header2.signature,
+					expect.any(Buffer),
+				)
+				.mockReturnValue(false);
+
+			await expect(pomCommand.execute(context)).rejects.toThrow(
+				'Invalid block signature for header 2',
+			);
+		});
+
+		it('should throw error if misbehaving account is not a delegate', async () => {
+			const updatedDelegateAccount = objects.cloneDeep(misBehavingDelegate);
+			updatedDelegateAccount.name = '';
```

Comment:
instead of emptying the name, it should use non-existing key for the delegateStore